### DISCUSSION TITLE
TopLevel Button Action

### DISCUSSION THREAD

**Discussion by JonathanBruner** [TomSchimansky/CustomTkinter#1237] [Q&A] [answered]
I am creating an app and was able to successfully create a TopLevel window, however inside that TopLevel I have a button and whenever I open the window that "command" immediately runs. How do set it to only run when the button is pressed. I didn't have this issue whenever I added buttons to my main window. I wonder if it's the way I'm calling the TopLevel window. I have attached my TopLevel window call as well as the TopLevel Class

```
    def new_user_setup(self):
        if self.new_user_setup_window is None or not self.new_user_setup_window.winfo_exists():
            self.new_user_setup_window = NewUserSetup(self)  # create window if its None or destroyed
            self.new_user_setup_window.focus()
        else:
            self.new_user_setup_window.focus()  # if window exists focus it
```

```
class NewUserSetup(customtkinter.CTkToplevel):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        # configure window
        self.title("New User Setup Console")
        self.geometry(f"{800}x{700}")

        self.get_users_button = customtkinter.CTkButton(self, text="User Export", command=app.get_users())
        self.get_users_button.grid(row=1, column=1, padx=20, pady=(10, 10))
```

**Comment by Wolf-SO** [comment, marked as answer]
@JonathanBruner This is because you *call* the function when you configure the button's action. Just remove the parentheses after `app.get_users`.

**Reply by JonathanBruner** [reply]
Thank you!
